src/test/java/de/retest/web/selenium/css/PredicateBuilderTest.java

```diff
@@ -0,0 +1,114 @@
+package de.retest.web.selenium.css;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.slf4j.LoggerFactory;
+
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.read.ListAppender;
+import de.retest.recheck.ui.descriptors.Element;
+
+@ExtendWith( MockitoExtension.class )
+@ExtendWith( SoftAssertionsExtension.class )
+public class PredicateBuilderTest {
+
+	private static final String CSS_SELECTOR_NOT_SUPPORTED =
+			"Unbreakable tests are not implemented for all CSS selectors.";
+	@Mock
+	private Element element;
+	@Mock
+	private Element otherElement;
+	private List<ILoggingEvent> logsList;

```

Comment:
Check out e.g. `DefaultRecheckIgnoreJsTest` in recheck, here we use [LogUnit](https://github.com/netmikey/logunit/) for this.

Reply:
Nice, I didn't know LogUnit before.